src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Validation/Narratives/NarrativeValidatorTests.cs

```diff
@@ -27,29 +31,39 @@ public NarrativeValidatorTests()
 
         [Theory]
         [MemberData(nameof(XssStrings))]
-        public void GivenAnInvalidNarrative_WhenProcessingAResource_ThenAValidationMessageIsCreated(string maliciousNarrative)
+        public void GivenAnInvalidNarrative_WhenProcessingAResource_ThenAValidationMessageWithAFhirPathIsCreated(string maliciousNarrative)
         {
             var defaultObservation = Samples.GetDefaultObservation().ToPoco<Observation>();
             defaultObservation.Text.Div = maliciousNarrative;
 
-            var result = _validator.Validate(
+            var resourceElement = defaultObservation.ToResourceElement();
+
+            IEnumerable<ValidationFailure> result = _validator.Validate(
                 new PropertyValidatorContext(
-                    new ValidationContext(defaultObservation.ToResourceElement()),
+                    new ValidationContext(resourceElement),
                     PropertyRule.Create<ResourceElement, ResourceElement>(x => x),
                     "Resource"));
 
-            Assert.NotEmpty(result);
+            List<ValidationFailure> validationFailures = result as List<ValidationFailure> ?? result.ToList();
+            Assert.NotEmpty(validationFailures);
+
+            var actualFhirPath = validationFailures.FirstOrDefault()?.PropertyName;
+            var expectedFhirPath = resourceElement.Instance.InstanceType + "." + KnownFhirPaths.ResourceNarrative;
+
+            Assert.Equal(expectedFhirPath, actualFhirPath);
         }
 
         [Theory]
         [MemberData(nameof(XssStrings))]
-        public void GivenAnInvalidNarrative_WhenProcessingABundle_ThenAValidationMessageIsCreated(string maliciousNarrative)
+        public void GivenAnInvalidNarrative_WhenProcessingABundle_ThenAValidationMessageWithAFhirPathIsCreated(string maliciousNarrative)
         {
             var defaultObservation = Samples.GetDefaultObservation().ToPoco<Observation>();
             defaultObservation.Text.Div = maliciousNarrative;
+            var observationInstance = defaultObservation.ToResourceElement().Instance;
```

Comment:
Perhaps just use `defaultObservation.TypeName` below instead of converting.
There is also a `[ResourceElement].InstanceType` that exists without needing to navigate to the Instance

Reply:
Oh cool, thanks. Fixed in 9c8d694.